### PR TITLE
feat(tauri): add `async` support

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Review by wusyong — COMMENTED** [review]
What's the reason all `Webview`s are changed to `WebviewMut`?
Is it because the lifetime issue?
If so, let me update the upstream a bit to make ease of that.
It could also reduce lots of mutex usages too I think.

File: cli/tauri.js/test/jest/fixtures/app/src-tauri/src/main.rs [hunk 1]
```diff
@@ -16,11 +16,13 @@ fn main() {
         )
         .unwrap();
       });
-      webview.eval("window.onTauriInit && window.onTauriInit()");
+      let _ = webview.dispatch(|w| {
+        w.eval("window.onTauriInit && window.onTauriInit()");
+      });
```
Comment (wusyong) [inline cli/tauri.js/test/jest/fixtures/app/src-tauri/src/main.rs:21]: I look around all `dispatch` calls and they all simpy calling `eval`.
Is it possible we just call `eval` instead?

**Comment by lucasfernog** [comment]
I had to change all webviews to WebviewMut because it is Send @wusyong

File: cli/tauri.js/test/jest/fixtures/app/src-tauri/src/main.rs (hunk 1 above)
Comment (lucasfernog) [reply cli/tauri.js/test/jest/fixtures/app/src-tauri/src/main.rs:21]: there's no "eval" on WebviewMut, so the dispatch is needed.

**Pull request merged**